### ISSUE TITLE
Login from NC login no longer automatic after update to Nextcloud 16

### ISSUE THREAD

**Issue by deusoz** [jsxc/jsxc#783]
### Expected behavior
When logging into Nextcloud, the jsxc app should login the user as well.
I have working instance Nextcloud 15.05 with non-updated jsxc @version 3.4.4 working fine and does just that.  

### Actual behavior
 For another Nextcloud instance on same server I updated NextCloud to 16.0 and jsxc to 3.4.5.  For that one when logging into Nextcloud using the normal Login button it quickly logs in, however the chat needs again to be logged in and using full domain JID.  user@mydomain.com that is in Prosody list. 

I installed fresh Nextcloud 16 on another server now running jsxc v3.4.5 from the first installation.
After login to Nextcloud using regular "Login" button, the page freezes with spinning "logging in". After a page refresh,  am logged in to Nextcloud fine, but must  "relogin" the Prosody user with the full domain as username.

Each instance has a [username] with [pw] for Nextcloud and same[username]@domain.com with exact same [pw] as a registered Prosody user.
Centos 7
Apache running behind Nginx

**Comment by sualko** [comment]
Can you post your javascript log? I looked at all commits and there is no change to the login process. Maybe you can send me some test credentials to klaus@jsxc.org.

**Comment by deusoz** [comment]
I think perhaps it is not so much your update, but rather the update of Nextcloud from 15 to 16 that broke it.  Very fortunately I have 3 installation on same server running Plesk/Centos7 all on the same Prosody server on same server.  I upgraded one to 16, but didn't check if login was broke until after I upgraded jsxc app to 3.4.5.  So it could have just as well been the NC upgrade that caused issue.

Here is javascript console from one NC login page that is working (NC15.05 / jsxc v3.4.4 )
Click "Login" and it logs into NC with user also logged into Prosody correctly...

`Variable serverType doesn't exist. It was created. jsxc.min.js:1:2062
Variable loginForm doesn't exist. It was created. jsxc.min.js:1:2062
Variable screenMediaExtension doesn't exist. It was created. jsxc.min.js:1:2062
Variable xmpp doesn't exist. It was created. jsxc.min.js:1:2062
Variable adminSettings doesn't exist. It was created. jsxc.min.js:1:2062
State changed to ESTABLISHING jsxc.min.js:1:2062
New connection jsxc.min.js:1:2062
CONNECTING: null jsxc.min.js:1:2062
CONNECTED: null jsxc.min.js:1:2062
Navigated to https://merlynda.cloud/index.php/login
XML Parsing Error: not well-formed
Location: 
Line Number 1, Column 168:
JQMIGRATE: Migrate is installed, version 1.4.0 core.js:7:552
Content Security Policy: The page’s settings blocked the loading of a resource at https://div.show/element (“frame-src”).
State changed to INITIATING jsxc.min.js:1:2062
State changed to PREVCONFOUND jsxc.min.js:1:2062
check master jsxc.min.js:1:2062
I am master. jsxc.min.js:1:2062
State changed to ESTABLISHING jsxc.min.js:1:2062
Try to attach jsxc.min.js:1:2062
SID: 8015b815-c9d8-45c2-96e2-e405e99f66ae jsxc.min.js:1:2062
ATTACHED: null jsxc.min.js:1:2062
UI State changed to INITIATING jsxc.min.js:1:2062
Waiting for server capabilities jsxc.min.js:1:2062
Request server capabilities jsxc.min.js:1:2062
DSA key loaded jsxc.min.js:1:2062
State changed to READY jsxc.min.js:1:2062
Discover muc service jsxc.min.js:1:2062
discover http upload service jsxc.min.js:1:2062
Load bookmarks from pubsub jsxc.min.js:1:2062
Roster loaded jsxc.min.js:1:2062
[webrtc] Could not update icon, because could not find jid for merlynda.cloud jsxc.min.js:1:2062
UI State changed to READY jsxc.min.js:1:2062
No photo provided jsxc.min.js:1:2062
XML Parsing Error: not well-formed
Location: 
Line Number 1, Column 168: `

This is the console log from broken instance on same server (NC16 JSXC 3.4.5)
Logs in to NC but not Prosody.  Must "relogin" from chat window and then needs full username user@domain to log into Prosody, then it's good to go...

`Variable serverType doesn't exist. It was created. jsxc.min.js:1:2062
Variable loginForm doesn't exist. It was created. jsxc.min.js:1:2062
Variable screenMediaExtension doesn't exist. It was created. jsxc.min.js:1:2062
Variable xmpp doesn't exist. It was created. jsxc.min.js:1:2062
Variable adminSettings doesn't exist. It was created. jsxc.min.js:1:2062
State changed to ESTABLISHING jsxc.min.js:1:2062
New connection jsxc.min.js:1:2062
CONNECTING: null jsxc.min.js:1:2062
CONNECTED: null jsxc.min.js:1:2062
Navigated to https://cloud.deusoz.com/index.php/login?clear=1
XML Parsing Error: not well-formed
Location: 
Line Number 1, Column 168:
JQMIGRATE: Migrate is installed, version 1.4.1 main.js:278:531
Content Security Policy: The page’s settings blocked the loading of a resource at https://div.show/element (“frame-src”).
I turned out your storage and deleted 11 entries. jsxc.min.js:1:2062
State changed to INITIATING jsxc.min.js:1:2062
Try to relogin jsxc.min.js:1:2062
Logout was forced or I found no valid jid jsxc.min.js:1:2062
State changed to SUSPEND jsxc.min.js:1:2062
[WARN] Unable to create user prefix `

**Comment by sualko** [comment]
Looks like, the connection is established first and than forgotten. Can you enable [debugging](https://github.com/jsxc/jsxc/wiki/Debugging) und post the log again? I need to see the xmpp stanzas. Otherwise send me some test credentials as mentioned above.

**Comment by deusoz** [comment]
### Environment
- **JSXC version:3.4.5
- **Host system and version: Nextcloud 
- **Browser vendor and version: Firefox v57 running on Ubuntu 18.04.2
- **Any browser plugins enabled?plugins-no; extensions-BitWardenPasswordManager, CorsEverywhere, DuckDuckGoPrivacy, SplitUpTabmanager, YoutubeVideoDownloader
- **XMPP server vendor and version:Prosody 0.11.2 Lua 5.1
- **Is your XMPP server working with other clients as expected? yes with Nextcloud v15 and jsxcv3.4.4

### Logs

### error

`XML Parsing Error: not well-formed
Location: 
Line Number 1, Column 168:
Content Security Policy: The page’s settings blocked the loading of a resource at https://div.show/element (“frame-src”).`

### complete console log with debug on
`Variable serverType doesn't exist. It was created. jsxc.min.js:1:2062
Variable loginForm doesn't exist. It was created. jsxc.min.js:1:2062
Variable screenMediaExtension doesn't exist. It was created. jsxc.min.js:1:2062
Variable xmpp doesn't exist. It was created. jsxc.min.js:1:2062
Variable adminSettings doesn't exist. It was created. jsxc.min.js:1:2062
State changed to ESTABLISHING jsxc.min.js:1:2062
New connection jsxc.min.js:1:2062
CONNECTING: null jsxc.min.js:1:2062
CONNECTED: null jsxc.min.js:1:2062
Navigated to https://cloud.deusoz.com/index.php/login?clear=1
XML Parsing Error: not well-formed
Location: 
Line Number 1, Column 168:
Content Security Policy: Ignoring duplicate source 'unsafe-inline'
Content Security Policy: Couldn’t parse invalid host 'unsafe-inline'
Content Security Policy: Directive ‘child-src’ has been deprecated. Please use directive ‘worker-src’ to control workers, or directive ‘frame-src’ to control frames respectively.
JQMIGRATE: Migrate is installed, version 1.4.1 main.js:278:531
Content Security Policy: The page’s settings blocked the loading of a resource at https://div.show/element (“frame-src”).
I turned out your storage and deleted 11 entries. jsxc.min.js:1:2062
State changed to INITIATING jsxc.min.js:1:2062
Try to relogin jsxc.min.js:1:2062
Logout was forced or I found no valid jid jsxc.min.js:1:2062
State changed to SUSPEND jsxc.min.js:1:2062
[WARN] Unable to create user prefix
jsxc.min.js:1:2062
New search handler registered search.js:67:12
The escapeHTML library is deprecated! It will be removed in nextcloud 19.
main.js:750:58
HERE I AM script:1:39
[WARN] Unable to create user prefix jsxc.min.js:1:2062
The escapeHTML library is deprecated! It will be removed in nextcloud 19.
main.js:750:58
The select2 library is deprecated! It will be removed in nextcloud 19.
main.js:750:58
The humanFileSize library is deprecated! It will be removed in nextcloud 19.
main.js:750:58
querying Whats New data was successful: nocontent main.js:223:1268
undefined main.js:223:1328
[WARN] Unable to create user prefix`

#### XMPP

no

#### Host

no

**Comment by deusoz** [comment]
I swiched to Ejabberd and now it is behaving better.

**Comment by deusoz** [comment]
I changed to ejabberd and it's auto logging in now.

**Comment by deusoz** [comment]
I set up two identical servers, Centos7, no Plesk, ejabberd,  
One I cloned Nextcloud 15 stable and downloaded Jsxc from appstore.
Other I cloned in nextcloud 16 stable and again Jsxc from appstore.
The version 15 definitely autologs in better.  The version 16 auto-login works only sometimes,  then loses the roster for that user sometimes, it seems buggy to be sure on NC16 whereas not buggy on NC15.

**Comment by deusoz** [comment]
Here is console log of NC 16 trying to log in... but never does.  

Content Security Policy: Ignoring duplicate source 'unsafe-inline'
Content Security Policy: Couldn’t parse invalid host 'unsafe-inline'
Content Security Policy: Directive ‘child-src’ has been deprecated. Please use directive ‘worker-src’ to control workers, or directive ‘frame-src’ to control frames respectively.
JQMIGRATE: Migrate is installed, version 1.4.1 main.js:278:531
I turned out your storage and deleted 12 entries. jsxc.min.js:1:2062
State changed to INITIATING jsxc.min.js:1:2062
Try to relogin jsxc.min.js:1:2062
Logout was forced or I found no valid jid jsxc.min.js:1:2062
State changed to SUSPEND jsxc.min.js:1:2062
[WARN] Unable to create user prefix
jsxc.min.js:1:2062
New search handler registered search.js:67:12
The escapeHTML library is deprecated! It will be removed in nextcloud 19.
main.js:750:58
TypeError: Strophe.getBareJidFromJid(...) is null
jsxc.min.js:1:13312
    jidToBid https://aho.we1.cloud/apps/ojsxc/js/jsxc/jsxc.min.js?v=8c9b802e:1
    onSlave https://aho.we1.cloud/apps/ojsxc/js/jsxc/jsxc.min.js?v=8c9b802e:1
    onStorage https://aho.we1.cloud/apps/ojsxc/js/jsxc/jsxc.min.js?v=8c9b802e:1
I am the slave. jsxc.min.js:1:2062
The escapeHTML library is deprecated! It will be removed in nextcloud 19.
main.js:750:58
The select2 library is deprecated! It will be removed in nextcloud 19.
main.js:750:58
Content Security Policy: The page’s settings blocked the loading of a resource at https://div.show/element (“frame-src”).
The humanFileSize library is deprecated! It will be removed in nextcloud 19.
main.js:750:58
querying Whats New data was successful: nocontent main.js:223:1268
undefined main.js:223:1328
[WARN] Unable to create user prefix
jsxc.min.js:1:2062
check master
jsxc.min.js:1:2062
check master jsxc.min.js:1:2062
check master jsxc.min.js:1:2062
check master

**Comment by deusoz** [comment]
Then "check master" keeps showing every four seconds

**Comment by deusoz** [comment]
Perhaps it is the update of JQMigrate of 1.4.0 to 1.4.1 

on the non-working server (NC16), and right before it clearly breaks, we get debug messages

```
JQMIGRATE: Migrate is installed, version 1.4.1 main.js:278:531
I turned out your storage and deleted 12 entries. jsxc.min.js:184:18
State changed to INITIATING jsxc.min.js:184:18
Try to relogin
```
while on the working version , same setup same except NC 15
```
JQMIGRATE: Migrate is installed, version 1.4.0 core.js:7:552
State changed to INITIATING jsxc.min.js:184:18
State changed to PREVCONFOUND jsxc.min.js:184:18
check master jsxc.min.js:184:18
New search handler registered search.js:67:12
I am master.
```

**Comment by deusoz** [comment]
going by last theory.. have solved the problem by commenting out line on line 231 of jsxc.js file
`   init: function(options) {
     // jsxc.runMigrations();`

Now my Nextcloud 16 is logging in the chat user with every NC login.  Ah relief.

**Comment by deusoz** [comment]
And this _did_ break the finding of the buddy list.   It logs in automatically with line commented out, but does not find buddy list.  Went back uncommented line and refreshed, again, I got logged out of chat, but then when I logged back in via "relogin" in chat window using full JID with domain name, the buddy list comes back.  So commenting out line above solves auto-login but then loses buddy list.

**Comment by deusoz** [comment]
Just now confirmed again. if comment out line above, on logout/login it auto-logs the chat user.  This last time it retained the roster as well.  I will investigate further.

**Comment by deusoz** [comment]
One mistake I made is that all of my JSXC versions say 3.4.4  
I was fairly certain when I upgraded to NC16 it told me I was downloading 3.4.5 but the files and apps list still say 3.4.4

**Comment by deusoz** [comment]
local storage is not persisting after logout for NC16, while it _is_ persisting for NC15

**Comment by deusoz** [comment]
So that does seem to be the issue with NC16... 
If I comment out line 47 in core/js/login.js   then the chat app works as it did with NC15

```
	if (clearParamRegex.test(window.location.href)) {
		//window.localStorage.clear();
		window.sessionStorage.clear();
	}
```

**Comment by sualko** [comment]
Wtf. After every logout Nextcloud clears all client side stores. This requires an upstream change. Thanks for the investigation and your time.

**Comment by sualko** [comment]
I opened a issue https://github.com/nextcloud/server/issues/15943

**Comment by sualko** [comment]
Btw the login is still working for me, but you can try to add this snippet to the top of `osjxc/js/ojsxc.js`:

```
window.localStorage.clear = function() {
  Object.keys(localStorage).filter(function(key) {
    return key.indexOf('jsxc') !== 0;
  }).forEach(function(key) {
    localStorage.removeItem(key)
  });
};
```

**Comment by deusoz** [comment]
So with localStorage not getting cleared out, login works as expected.

However NC16  still now logs _out_ chat user after a short while on my two NC16 instances as described above, whereas not on NC15 on identical setups,

 I now have 3 Centos7/apache/nginx servers running with their own chat servers, 1 with Prosody, 2 with Ejabberd.  On one of the ejabberds I have NC15, other has NC16.  the Prosody server has many domains, I have 2 instances running NC15 and one running NC16 on same server all on same Prosody server. 

 The Ejabberd running NC15 at "idle" but with logged in chat user, produces absolutely no xmpp related messages in console and just stays logged in, NC16 on identical server has constant messages related to jsxc / xmpp.  (the Prosody server NC15 instances at idle does show  xmpp error, debugs and bosh url console messages at idle, but never loses connection, the NC16 on Prosody eventually always has similar breakdown as shown below )

Here's my Ejabberd/NC15 (from my Firefox on Ubuntu) console, all messages on, idling with chat on and logged in.  All the while the storage number for jsxc:alive[n]:master is incrementing every second as expected.
`XHRGEThttps://us1.db5d.net/index.php/csrftoken
[HTTP/2.0 200 OK 956ms]

XHRGEThttps://us1.db5d.net/index.php/csrftoken
[HTTP/2.0 200 OK 4908ms]

XHRGEThttps://us1.db5d.net/index.php/csrftoken
[HTTP/2.0 200 OK 2418ms]

XHRGEThttps://us1.db5d.net/index.php/csrftoken
[HTTP/2.0 200 OK 989ms]

XHRGEThttps://us1.db5d.net/index.php/csrftoken
[HTTP/2.0 200 OK 3013ms]

XHRGEThttps://us1.db5d.net/index.php/csrftoken
[HTTP/2.0 200 OK 4277ms]

XHRGEThttps://us1.db5d.net/index.php/csrftoken
[HTTP/2.0 200 OK 840ms]

XHRGEThttps://us1.db5d.net/index.php/csrftoken
[HTTP/2.0 200 OK 1941ms]

XHRGEThttps://us1.db5d.net/index.php/csrftoken
[HTTP/2.0 200 OK 2607ms]`

All you see is csrf for hours in the console. Chat stays logged in and user is "online" hours later.

But for the NC16 on identical setup with Ejabberd, we get a lot of errors about misformed xml and then eventually it loses connection. Here is console showing connection loss from NC 16 with Ejabberd. You'll notice there are a few extra debug calls that i put in there, such as `THIS IS STATUS DISCONNECTED coming from strophe jsxc.min.js:1312:24`  which never happens on the NC15 instances. 

`XML Parsing Error: not well-formed
Location: 
Line Number 1, Column 120:
XHRGEThttps://aho.we1.cloud/index.php/avatar/db5d/36?requesttoken=%2F7UwTjZ2%2FfmnoUa2GbmsJQJNvMrOEESrPoAF5UKBkM0%3D%3AjsV7NHECtbTBwgHmK96DUTp824T9QXXgbM5wqHHS170%3D
[HTTP/2.0 201 Created 416ms]

XHRPOSThttps://aho.we1.cloud/bosh
[HTTP/2.0 200 OK 243ms]

Presence (stuart@cloud.deusoz.com/_JZOQeii): online jsxc.min.js:184:18
Presence (db5d@we2.cloud/83253229891271782634611): online jsxc.min.js:184:18
XML Parsing Error: not well-formed
Location: 
Line Number 1, Column 120:

Variable db5d@36 doesn't exist. It was created. jsxc.min.js:184:18
XHRPOSThttps://aho.we1.cloud/bosh
[HTTP/2.0 404 Not Found 60410ms]

GEThttps://aho.we1.cloud/index.php/avatar/db5d/36?requesttoken=%2F7UwTjZ2%2FfmnoUa2GbmsJQJNvMrOEESrPoAF5UKBkM0%3D%3AjsV7NHECtbTBwgHmK96DUTp824T9QXXgbM5wqHHS170%3D
[HTTP/2.0 304 Not Modified 550ms]

DISCONNECTING: null jsxc.min.js:184:18
DISCONNECTED: undefined jsxc.min.js:184:18
THIS IS STATUS DISCONNECTED coming from strophe jsxc.min.js:1312:24
Try to relogin jsxc.min.js:184:18
db5d@aho.we1.cloud/73431513531199954832370 jsxc.min.js:184:18
db5d@aho.we1.cloud jsxc.min.js:184:18
Object { url: "/bosh", jid: "db5d@aho.we1.cloud", domain: "aho.we1.cloud", password: null, sid: null, rid: null, overwrite: true, onlogin: null, resource: "", username: "db5d" }
jsxc.min.js:184:18
ELSE jsxc.min.js:184:18
NOT! jsxc.min.js:553:20
Object { url: "/bosh", jid: "db5d@aho.we1.cloud", domain: "aho.we1.cloud", password: null, sid: null, rid: null, overwrite: true, onlogin: null, resource: "", username: "db5d" }
jsxc.min.js:554:20
loadSettingsAllKnowing = false jsxc.min.js:184:18
I am not able to relogin jsxc.min.js:184:18
disconnected jsxc.min.js:184:18
State changed to SUSPEND jsxc.min.js:184:18
XML Parsing Error: mismatched tag. Expected: </hr>.
Location: https://aho.we1.cloud/bosh
Line Number 6, Column 3: bosh:6:3
XHRGEThttps://aho.we1.cloud/index.php/csrftoken`

``

This all being said,  This is how I got these servers working and not working :>), not to say I'm completely 100% sure I have the optimum nginx setup and configs for each chat server with the SSL certificates, tls settings, etc, which are different for both.  But the all work server to server TLS encrypted so I'm doing something right.

**Comment by deusoz** [comment]
In all cases above, when one gets logged out, the "relogin" button works to quickly log back in.  And with the short username only, no domain needed as I had reported earlier.

**Comment by deusoz** [comment]
I may have some https nginx issues, not sure how suddenly the /bosh is a 404 when a second ago it was ok
`XHRPOSThttps://aho.we1.cloud/bosh
[HTTP/2.0 200 OK 243ms]

Presence (stuart@cloud.deusoz.com/_JZOQeii): online jsxc.min.js:184:18
Presence (db5d@we2.cloud/83253229891271782634611): online jsxc.min.js:184:18
XML Parsing Error: not well-formed
Location: 
Line Number 1, Column 120:

Variable db5d@36 doesn't exist. It was created. jsxc.min.js:184:18
XHRPOSThttps://aho.we1.cloud/bosh

POSThttps://aho.we1.cloud/bosh
[HTTP/2.0 404 Not Found 60410ms]`

The first one and many preceding sometimes has:
`XHRPOSThttps://aho.we1.cloud/bosh
[HTTP/2.0 200 OK 243ms]

	
Response payload	

1

<body xmlns='http://jabber.org/protocol/httpbind'><iq xml:lang='en' to='db5d@aho.we1.cloud/73431513531199954832370' from='db5d@aho.we1.cloud' type='result' id='513fb20f-d612-4dfc-961e-e68ca723394a:sendIQ' xmlns='jabber:client'><query xmlns='http://jabber.org/protocol/disco#info'><identity type='pep' category='pubsub'/><identity type='registered' category='account'/><feature var='http://jabber.org/protocol/disco#info'/><feature var='http://jabber.org/protocol/disco#items'/><feature var='vcard-temp'/><feature var='urn:xmpp:push:0'/><feature var='urn:xmpp:bookmarks-conversion:0'/><feature var='urn:xmpp:pep-vcard-conversion:0'/><feature var='msgoffline'/><feature var='http://jabber.org/protocol/offline'/><feature var='http://jabber.org/protocol/pubsub'/><feature var='http://jabber.org/protocol/pubsub#create-nodes'/><feature var='http://jabber.org/protocol/pubsub#auto-create'/><feature var='http://jabber.org/protocol/pubsub#auto-subscribe'/><feature var='http://jabber.org/protocol/pubsub#delete-nodes'/><feature var='http://jabber.org/protocol/pubsub#delete-items'/><feature var='http://jabber.org/protocol/pubsub#filtered-notifications'/><feature var='http://jabber.org/protocol/pubsub#modify-affiliations'/><feature var='http://jabber.org/protocol/pubsub#outcast-affiliation'/><feature var='http://jabber.org/protocol/pubsub#persistent-items'/><feature var='http://jabber.org/protocol/pubsub#publish'/><feature var='http://jabber.org/protocol/pubsub#publish-options'/><feature var='http://jabber.org/protocol/pubsub#purge-nodes'/><feature var='http://jabber.org/protocol/pubsub#retract-items'/><feature var='http://jabber.org/protocol/pubsub#retrieve-affiliations'/><feature var='http://jabber.org/protocol/pubsub#retrieve-items'/><feature var='http://jabber.org/protocol/pubsub#retrieve-subscriptions'/><feature var='http://jabber.org/protocol/pubsub#subscribe'/><feature var='http://jabber.org/protocol/commands'/></query></iq><presence to='db5d@aho.we1.cloud/73431513531199954832370' from='stuart@cloud.deusoz.com/_JZOQeii' xmlns='jabber:client'><c xmlns='http://jabber.org/protocol/caps' hash='sha-1' ver='UEMHBlT9XVU7FUwfm/6rl815MUE=' node='http://jsxc.org/'/><delay xmlns='urn:xmpp:delay' from='cloud.deusoz.com' stamp='2019-06-14T21:52:38Z'/><x xmlns='vcard-temp:x:update'/></presence><presence xml:lang='en' to='db5d@aho.we1.cloud/73431513531199954832370' from='db5d@we2.cloud/83253229891271782634611' xmlns='jabber:client'><c xmlns='http://jabber.org/protocol/caps' hash='sha-1' node='http://jsxc.org/' ver='UEMHBlT9XVU7FUwfm/6rl815MUE='/><x xmlns='vcard-temp:x:update'/><delay xmlns='urn:xmpp:delay' from='db5d@we2.cloud/83253229891271782634611' stamp='2019-06-14T16:52:01.902153Z'/></presence></body>`

Then some milliseconds later there's a 404 .    Maybe I need to up the timeout for nginx, not sure.  NC15 server has exact same nginx/httpd/ssl config and is not making a peep.

**Comment by deusoz** [comment]
Ok,  I added some directives and now the NC16 Ejabberd console has quieted down
added this to the /bosh proxy config

proxy_connect_timeout       600;
proxy_send_timeout          600;
proxy_read_timeout          600;
send_timeout                600;

did this
Try raising max_execution_time setting in php.ini file (CentOS path is /etc/php.ini):
max_execution_time = 300

am running php-fmp so set this as well
/etc/php-fpm.d/www.conf
request_terminate_timeout = 300

and now, it's all quiet, no errors in console , ok so nc16 is less forgiving of incomplete nginx / php configs
`XHRPOSThttps://aho.we1.cloud/bosh
[HTTP/2.0 200 OK 60206ms]

XHRPOSThttps://aho.we1.cloud/bosh
[HTTP/2.0 200 OK 60191ms]

XHRPOSThttps://aho.we1.cloud/bosh
[HTTP/2.0 200 OK 60188ms]

XHRPOSThttps://aho.we1.cloud/bosh
[HTTP/2.0 200 OK 60188ms]

XHRPOSThttps://aho.we1.cloud/bosh
[HTTP/2.0 200 OK 60345ms]

XHRPOSThttps://aho.we1.cloud/bosh`

each one is very reassuringly stating
`<body xmlns='http://jabber.org/protocol/httpbind'/>`

**Comment by deusoz** [comment]
OMG I fixed it, LOL thanks for this great app Klaus :D

**Comment by sualko** [comment]
> I fixed it

:tada: 

> hanks for this great app 

No problem :wink:

**Comment by deusoz** [comment]
Hi Klaus,
Today I did another installation straight from the latest stable16 NC server.  The ONLY method that worked for me to be able to login chat user with Nextcloud user login was still to comment out the line 47 of core/js/login.js  .    I first tried adding code you suggested to beginning  of the osjxc.js, but that didn't fix issue.  Only commenting out this line:  
//      window.localStorage.clear();
seems to work for initial login to also login the chat user.
Best regards,
Stuart
